src/pygeoprocessing/routing/routing.pyx

```diff
@@ -2839,7 +2838,7 @@ def distance_to_channel_mfd(
                         pixel.value += flow_dir_weight * (
                             weight_val + n_distance)
 
-                    if preempted:
+                    if preempted or pixel.value == distance_nodata:

```

Comment:
what does `preempted` represent here? it's `1` when this pixel hasn't been visited before?

Reply:
the algorithm above works by finding a pixel that has an undefined distance to stream value, then attempts to work that out by walking downstream. if one of the flow directions is undefined we save off what direction we were considering, push that pixel on the stack, then push the uncalculated downstream pixel on the stack. in that case the algorithm was "preempted" because it wasn't able to finish calculating the distance to stream for the pixel. is that okay?